tests/scripts/deploy_admission_controller.sh

```diff
@@ -5,22 +5,52 @@ set -eo pipefail
 
 # Set our known directories and parameters.
 BASE_DIR=$(cd "$(dirname "$0")"; pwd)
+CERT_VERSION="v1.2.0"
 
 # Set the variables for webhook. Any change here for namespace and service name should be done in pkg/operator/admission/init.go also
 [ -z "${NAMESPACE}" ] && NAMESPACE="rook-ceph"
 WEBHOOK_CONFIG_NAME="rook-ceph-webhook"
 SERVICE_NAME="rook-ceph-admission-controller"
 
-INSTALL_SELF_SIGNED_CERT=true
 echo "$BASE_DIR"
 
-if [ "${INSTALL_SELF_SIGNED_CERT}" == true ]; then
-	"${BASE_DIR}"/webhook-create-signed-cert.sh --namespace ${NAMESPACE}
+kubectl apply -f https://github.com/jetstack/cert-manager/releases/download/$CERT_VERSION/cert-manager.yaml
+timeout 150 sh -c 'until [ $(kubectl -n cert-manager get pods --field-selector=status.phase=Running|grep -c ^cert-) -eq 3 ]; do sleep 1; done'
+if [ $? -ne 0 ]; then
```

Comment:
Since you have `-e`  you don't need to check the return code, do you?

Reply:
thanks for pointing, while doing changes I thought if commands failed it will exit immediately. But all resource of cert-manager still will be there. So I have removed `-e` and deleting the resources if commands fail.

Reply:
How do you delete them? How do you cleanup?

Reply:
` kubectl delete -f https://github.com/jetstack/cert-manager/releases/download/$CERT_VERSION/cert-manager.yaml`

Reply:
Ok, is this the only thing we want to cleanup?

Reply:
is the one I know but there might be other things to cleanup too.

Reply:
but I think till that step(creating cert-manager) this is the only thing we need to clean.

Reply:
How about we add a `cleanup()` function to trap an any error and then cleanup all resources?

The code will look like:

```bash
cleanup() {
  kubectl delete -f https://github.com/jetstack/cert-manager/releases/download/$CERT_VERSION/cert-manager.yaml || true
# remove the selfsigned-issuer
# remove serving-cert
# remove webhook-config.yaml
}
trap "cleanup" EXIT SIGINT ERR
```

With that you can re-add the `set -e` and remove the `if [ $? -ne 0 ]; then`, ultimately we will have a nicer cleanup of the execution, especially if a retry is involved.

Reply:
Added.

Reply:
I removed `EXIT` because the trap was called every time even if there is no error.